terraform/modules/front_door/waf.tf

```diff
@@ -152,4 +157,97 @@ resource "aws_wafv2_web_acl" "this" {
       sampled_requests_enabled   = true
     }
   }
+
+  rule {
+    name     = "login-ip-rate-limit"
+    priority = 7
+
+    action {
+      block {
+        custom_response {
+          response_code = 429
+        }
+      }
+    }
+
+    statement {
+      rate_based_statement {
+        aggregate_key_type = "IP"
+        limit              = 100
+
+        scope_down_statement {
+          regex_pattern_set_reference_statement {
+            arn = aws_wafv2_regex_pattern_set.waf_rate_limit_urls.arn
+
+            field_to_match {
+              uri_path {}
+            }
+
+            text_transformation {
+              priority = 0
+              type     = "URL_DECODE"
+            }
+          }
+        }
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "waf-login-ip-rate-limit"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "overall-ip-rate-limit"
+    priority = 8
+
+    action {
+      block {
+        custom_response {
+          response_code = 429
+        }
+      }
+    }
+
+    statement {
+      rate_based_statement {
+        aggregate_key_type = "IP"
+        limit              = 2000

```

Comment:
I think this a reasonable limit looking at prod on logit logs. Here's a profile of an IP address I found, that looked to be high load user pretty recently. They made a maximum of ~280 requests/min => ~1400 requests/5mins, so 2000 is a bit of headroom on this.

![image](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data-infrastructure/assets/109098469/14ce11a5-5677-4182-8a2e-cdda76cf08ed)

Reply:
Just to clarify, this was a high load user looking at all usage ever on prod

Reply:
Added a note about this in functionality regression testing ticket [CLDC-2738](https://dluhcdigital.atlassian.net/browse/CLDC-2738)